src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs

```diff
@@ -0,0 +1,707 @@
+            httpResponse = await this.HttpClient.SendAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            if (shouldTrace)
+            {
+                ServiceClientTracing.ReceiveResponse(invocationId, httpResponse);
+            }
+
+            HttpStatusCode statusCode = httpResponse.StatusCode;
+            cancellationToken.ThrowIfCancellationRequested();
+            string responseContent = null;
+
+            if ((int)statusCode != 200 && (int)statusCode != 400 && (int)statusCode != 404 && (int)statusCode != 409 && (int)statusCode != 422)
+            {
+                var ex = new HttpOperationException(string.Format("Operation returned an invalid status code '{0}'", statusCode));
+                if (httpResponse.Content != null)
+                {
+                    responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                }
+                else
+                {
+                    responseContent = string.Empty;
+                }
+
+                ex.Request = new HttpRequestMessageWrapper(httpRequest, requestContent);
+                ex.Response = new HttpResponseMessageWrapper(httpResponse, responseContent);
+                if (shouldTrace)
+                {
+                    ServiceClientTracing.Error(invocationId, ex);
+                }
+
+                httpRequest.Dispose();
+                if (httpResponse != null)
+                {
+                    httpResponse.Dispose();
+                }
+
+                throw ex;
+            }
+
+            // Create Result
+            var result = new HttpOperationResponse<object>();
+            result.Request = httpRequest;
+            result.Response = httpResponse;
+
+            // Deserialize Response
+            if ((int)statusCode == 200)
+            {
+                responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                try
+                {
+                    result.Body = SafeJsonConvert.DeserializeObject<ProductListModel>(responseContent, this.DeserializationSettings);
+                }
+                catch (JsonException ex)
+                {
+                    httpRequest.Dispose();
+                    if (httpResponse != null)
+                    {
+                        httpResponse.Dispose();
+                    }
+
+                    throw new SerializationException("Unable to deserialize the response.", responseContent, ex);
+                }
+            }
+
+            // Deserialize Response
+            if ((int)statusCode == 400)
+            {
+                responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                try
+                {
+                    result.Body = SafeJsonConvert.DeserializeObject<IDictionary<string, IList<string>>>(responseContent, this.DeserializationSettings);
```

Comment:
Are you sure this will deserialiaze?

Reply:
Removed IList and using string. Now able to deserialize properly